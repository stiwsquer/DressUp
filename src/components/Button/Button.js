import React from "react";
import { Link, useHistory } from "react-router-dom";
import "./style.scss";
 
export default function Button({
  children,
  type,
  nameOfTheClass,
  onClick,
  linkTo,
}) {

  return (
    <Link to={linkTo}>
      <button onClick={onClick} className={nameOfTheClass} type={type}>
        {children}
      </button>
    </Link>
  );
}
