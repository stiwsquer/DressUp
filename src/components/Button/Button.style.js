import styled from "styled-components";

export const StyledButton = styled.button`
  background: #202020;
  color: white;
  border: none;
  outline: none;
  padding: 1rem 3rem;
  letter-spacing: 0.15rem;
  font-weight: 900;
  cursor: pointer;
  &:hover {
    opacity: 0.7;
  }
`;
