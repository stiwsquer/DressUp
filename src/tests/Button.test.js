import { BrowserRouter as Router } from "react-router-dom";
import Button from "../components/Button/Button";

it("renders Button component without crashing", () => {
  shallow(
    <Router>
      <Button linkTo="#" />
    </Router>
  );
});

// 

it("simulates click events", () => {
  const onButtonClick = sinon.spy();
  const buttonWrapper = mount(
    <Router>
      <Button
        linkTo="#"
        type="submit"
        nameOfTheClass="submitButton"
        onClick={onButtonClick}
      />
    </Router>
  );
  buttonWrapper.find("button").simulate("click");
  // expect(onButtonClick).toHaveProperty("callCount", 1);
  expect(onButtonClick.calledOnce).toBe(true);
});

it("renders children when passed in", () => {
  const wrapper = shallow(
    <Router>
      {" "}
      <Button linkTo="#">
        <div className="unique" />
      </Button>
    </Router>
  );
  expect(wrapper.contains(<div className="unique" />)).toEqual(true);
});
it("Button snapshot - shallow", () => {
  const tree = shallow(
    <Router>
      <Button
        linkTo="#"
        type="submit"
        nameOfTheClass="submitButton"
        onClick={() => {}}
      />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});

it("Button snapshot - render", () => {
  const tree = render(
    <Router>
      <Button
        linkTo="#"
        type="submit"
        nameOfTheClass="submitButton"
        onClick={() => {}}
      />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});

it("Button snapshot - mount", () => {
  const tree = mount(
    <Router>
      <Button
        linkTo="#"
        type="submit"
        nameOfTheClass="submitButton"
        onClick={() => {}}
      />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});
