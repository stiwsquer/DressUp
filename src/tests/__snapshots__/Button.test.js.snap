// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot - mount 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <button
    className="submitButton"
    onClick={[Function]}
    type="submit"
  />
</a>
`;

exports[`Button snapshot - render 1`] = `
<a
  href="/"
>
  <button
    class="submitButton"
    type="submit"
  />
</a>
`;

exports[`Button snapshot - shallow 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Button
    linkTo="#"
    nameOfTheClass="submitButton"
    onClick={[Function]}
    type="submit"
  />
</Router>
`;
